src/RandomProgramGenerator.cpp

```diff
@@ -831,6 +832,17 @@ main(int argc, char **argv)
 			continue;
 		}
 
+		if (strcmp (argv[i], "--trans-memory-atomic") == 0) {
+			CGOptions::trans_memory_atomic(true);
+			CGOptions::volatiles(false);
```

Comment:
Why do you set `volatiles`?

Reply:
Transaction block do not accept volatile variable in it.

Reply:
Ok, thanks for explanation.